libpromises/generic_agent.c

```diff
@@ -116,6 +138,236 @@ void MarkAsPolicyServer(EvalContext *ctx)
+    Buffer *expbuf = BufferNew();
+    ExpandScalar(ctx, NULL, "this", BufferData(filebuf), expbuf);
+    if (strstr(BufferData(expbuf), "/.json"))
+    {
+        Log(LOG_LEVEL_DEBUG, "Skipping augments file '%s' because it failed to expand the base filename, resulting in '%s'",
+            BufferData(filebuf),
+            BufferData(expbuf));
+    }
+    else
+    {
+        Log(LOG_LEVEL_DEBUG, "Searching for augments file '%s'", BufferData(expbuf));
+        if (FileCanOpen(BufferData(expbuf), "r"))
+        {
+            JsonElement* augment = ReadJsonFile(BufferData(expbuf), LOG_LEVEL_ERR);
+            if (NULL != augment )
+            {
+                if (JsonGetElementType(augment) != JSON_ELEMENT_TYPE_CONTAINER ||
+                    JsonGetContainerType(augment) != JSON_CONTAINER_TYPE_OBJECT)
+                {
+                    Log(LOG_LEVEL_ERR, "Invalid augments file contents in '%s', must be a JSON object", BufferData(expbuf));
+                }
+                else
+                {
+                    Log(LOG_LEVEL_VERBOSE, "Loaded augments file '%s', installing contents", BufferData(expbuf));
+
+                    JsonIterator iter = JsonIteratorInit(augment);
+                    const char *key;
+                    while ((key = JsonIteratorNextKey(&iter)))
+                    {
+                        if (0 == strcmp("vars", key))
+                        {
+                            // load variables
+                            JsonElement* vars = JsonExpandElement(ctx, JsonObjectGet(augment, key));
+
+                            if (NULL == vars ||
+                                JsonGetElementType(vars) != JSON_ELEMENT_TYPE_CONTAINER ||
+                                JsonGetContainerType(vars) != JSON_CONTAINER_TYPE_OBJECT)
+                            {
+                                Log(LOG_LEVEL_ERR, "Invalid augments vars in '%s', must be a JSON object", BufferData(expbuf));
+                            }
+                            else
+                            {
+                                JsonIterator iter = JsonIteratorInit(vars);
+                                const char *vkey;
+                                while ((vkey = JsonIteratorNextKey(&iter)))
+                                {
+                                    JsonElement *data = JsonObjectGet(vars, vkey);
+                                    if (JsonGetElementType(data) == JSON_ELEMENT_TYPE_PRIMITIVE)
+                                    {
+                                        char *value = JsonPrimitiveToString(data);
+                                        Log(LOG_LEVEL_VERBOSE, "Installing augments variable '%s.%s=%s' from file '%s'",
+                                            SPECIAL_SCOPE_DEF, vkey, value, BufferData(expbuf));
+                                        EvalContextVariablePutSpecial(ctx, SPECIAL_SCOPE_DEF, vkey, value, CF_DATA_TYPE_STRING, "source=augments");
```

Comment:
I think this should be ``source=augments_file`` in order to follow the same naming as in masterfiles.